jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -645,7 +645,7 @@ public void unpack (WebAppContext context) throws IOException
                         long lastModified;
                         // read last modified without caching a war file
                         // so that webapp can be redeployed on Windows
-                        try (Resource r = Resource.newResource("jar:" + web_app + "!/", false)) {
+                        try (Resource r = Resource.newResource(web_app.getURL().toExternalForm(), false)) {
```

Comment:
`URL.toExternalForm()` does not escape properly, please use the URI version.

Also, why are we doing it this way?
Wouldn't it just be ...

``` java
long lastModified = web_app.lastModified();
```

... without the extra `Resource` instantiation?